l1-contracts/contracts/bridge/asset-router/L2AssetRouter.sol

```diff
@@ -54,43 +70,68 @@ contract L2AssetRouter is IL2AssetRouter, Initializable {
-        L1_CHAIN_ID = _l1ChainId;
+    constructor(
+        uint256 _l1ChainId,
+        uint256 _eraChainId,
+        address _l1AssetRouter,
+        address _legacySharedBridge
+    ) AssetRouterBase(_l1ChainId, _eraChainId, IBridgehub(L2_BRIDGEHUB_ADDR), L2_BASE_TOKEN_SYSTEM_CONTRACT_ADDR) {
         L2_LEGACY_SHARED_BRIDGE = _legacySharedBridge;
         if (_l1AssetRouter == address(0)) {
             revert EmptyAddress();
         }
-
         l1AssetRouter = _l1AssetRouter;
 
         _disableInitializers();
     }
 
-    /// @dev Used to set the assedAddress for a given assetId.
-    /// @dev Will be used by ZK Gateway
-    function setAssetHandlerAddress(bytes32 _assetId, address _assetAddress) external onlyL1AssetRouter {
-        assetHandlerAddress[_assetId] = _assetAddress;
-        emit AssetHandlerRegistered(_assetId, _assetAddress);
+    ///  @inheritdoc IL2AssetRouter
+    function setAssetHandlerAddress(
+        uint256 _originChainId,
+        bytes32 _assetId,
+        address _assetAddress
+    ) external override onlyAssetRouterCounterpart(_originChainId) {
+        _setAssetHandlerAddress(_assetId, _assetAddress);
+    }
+
+    /// @inheritdoc IAssetRouterBase
+    function setAssetHandlerAddressThisChain(
+        bytes32 _assetRegistrationData,
+        address _assetHandlerAddress
+    ) external override(AssetRouterBase) {
+        _setAssetHandlerAddressThisChain(L2_NATIVE_TOKEN_VAULT_ADDR, _assetRegistrationData, _assetHandlerAddress);
     }
 
+    /*//////////////////////////////////////////////////////////////
+                            INITIATTE DEPOSIT Functions
+    //////////////////////////////////////////////////////////////*/
+
+    /*//////////////////////////////////////////////////////////////
+                            Receive transaction Functions
+    //////////////////////////////////////////////////////////////*/
+
     /// @notice Finalize the deposit and mint funds
     /// @param _assetId The encoding of the asset on L2
     /// @param _transferData The encoded data required for deposit (address _l1Sender, uint256 _amount, address _l2Receiver, bytes memory erc20Data, address originToken)
-    function finalizeDeposit(bytes32 _assetId, bytes memory _transferData) public override onlyL1AssetRouter {
-        address assetHandler = assetHandlerAddress[_assetId];
-        if (assetHandler != address(0)) {
-            IL2AssetHandler(assetHandler).bridgeMint(L1_CHAIN_ID, _assetId, _transferData);
-        } else {
-            L2_NATIVE_TOKEN_VAULT.bridgeMint(L1_CHAIN_ID, _assetId, _transferData);
-            assetHandlerAddress[_assetId] = address(L2_NATIVE_TOKEN_VAULT);
-        }
+    // kl todo. FinalizeDeposit functions.
+    function finalizeDeposit(
+        // solhint-disable-next-line no-unused-vars
+        uint256,
+        bytes32 _assetId,
+        bytes calldata _transferData
+    ) public override onlyAssetRouterCounterpartOrSelf(L1_CHAIN_ID) {
```

Comment:
this function is not `payable`, so providing any non-zero l2Value for a bridge transaction is useless (i.e. the value used for `msgValue` of `IAssetHandler.bridgeBurn`

Reply:
yes, that will be for the weth bridges

Reply:
I understand that weth bridges are not supported, but considering that we already provide `msgValue` into the asset handlers, it makes sense to make these methods payable

Reply:
An alternative is to explicitly comment in the interface for asset handlers that the value passed into the `msgValue` is always 0

Reply:
Ok, currently it is the second case, so I will do that.